### ISSUE TITLE
Cannot call service from javascript

### ISSUE THREAD

**Issue by Barma-lej** [MTrab/landroid_cloud#223]
### Environment:
- Home Assistant version: 2022.7.5
- Landroid_cloud version: 2.2.2
- Operation system: Home Assistant Supervised

## Describe the bug

Hi, Malene. The other day I have tried to call service `setzone` from javascript ([Description](https://developers.home-assistant.io/docs/frontend/data/#hasscallservicedomain-service-data)). And get an error: _'NoneType' object has no attribute 'device_id'_

```javascript
this.hass.callService('landroid_cloud', 'setzone', { entity_id: 'vacuum.mower', zone: 1 });
this.hass.callService('landroid_cloud', 'setzone', { entity_id: 'vacuum.mower', zone: '1' });
```

Than i tried to change `raindelay`

```javascript
this.hass.callService('landroid_cloud', 'config', { entity_id: 'vacuum.mower', config: 90, });
this.hass.callService('landroid_cloud', 'config', { entity_id: 'vacuum.mower', config: '90', });

```

And again i got the same error

After that i decided to call `set_fan_speed` for my vacuum cleaner. And that works without any problem

```javascript
this.hass.callService('vacuum', 'set_fan_speed', { entity_id: 'vacuum.robi', fan_speed: 'Silent' });
```

Can you please check this [line](https://github.com/MTrab/landroid_cloud/blob/7851e70e2662ce74e5a1befd26367e7d238c4d81/custom_components/landroid_cloud/services.py#L118)

I'm not sure that's error in your integration, but i can't find errors in javascript too 😒

## Debug log

```text

Logger: homeassistant.components.websocket_api.http.connection
Source: custom_components/landroid_cloud/services.py:118
Integration: Home Assistant WebSocket API ([documentation](https://www.home-assistant.io/integrations/websocket_api), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+websocket_api%22))
First occurred: 19 июля 2022 г., 20:06:08 (4 occurrences)
Last logged: 00:04:39

[140448154679424] 'NoneType' object has no attribute 'device_id'
[140447691906960] 'NoneType' object has no attribute 'device_id'
[140447692391136] 'NoneType' object has no attribute 'device_id'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 193, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1713, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1750, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/config/custom_components/landroid_cloud/services.py", line 118, in async_call_landroid_service
    entity_registry.entities.get(entry).device_id
AttributeError: 'NoneType' object has no attribute 'device_id'

```

**Comment by MTrab** [comment]
I'm not familiar with the Javascript calls in HA yet (haven't ventured into Lovelace developing yet) - but.
If you view the YAML definition of calling a service, from the developer console, it would look like this, calling the service with an entity_id rather than device_id.

```
service: landroid_cloud.setzone
data:
  zone: '1'
target:
  entity_id: vacuum.back_left
```

How this would translate to javascript calls I don't know.

**Comment by Barma-lej** [comment]
Well, if i call a service with `device_id` it works:
```javascript
this.hass.callService('landroid_cloud', 'setzone', { device_id: 'b2433e6xxxxxxxxxxx978e42cd0749', zone: 1, });
```
Doesn't work with `entity_id` 

However, if i call a service for vacuum everithing works
```javascript
this.hass.callService('vacuum', 'set_fan_speed', { entity_id: 'vacuum.robi', fan_speed: 'Silent' });
```
this is what it looks like in the developer console
```yaml
service: vacuum.set_fan_speed
data:
  fan_speed: sw
target:
  entity_id: vacuum.robi
```

**Comment by MTrab** [comment]
Which model is your vacuum?

**Comment by Barma-lej** [comment]
Landroid Worx M500 WR141E

**Comment by MTrab** [comment]
Not your mower, your vacuum ;) Need to see what the difference could be, as the code is as it should be and it reacts when you set at target entity_id in the developer tools

**Comment by Barma-lej** [comment]
🙈 i'm sorry

Roborock S50
Attributes:
```
fan_speed_list: Silent, Standard, Medium, Turbo, Gentle, Auto
battery_level: 100
battery_icon: mdi:battery-charging-100
fan_speed: Turbo
status: Charging
timers: 
- enabled: false
  cron: 37 15 * * 0,6
  next_schedule: '2022-07-24T13:37:00+00:00'

friendly_name: Robi
supported_features: 14204
```

**Comment by MTrab** [comment]
Which integration are you using for that one?

**Comment by Barma-lej** [comment]
[Xiaomi miio Github](https://github.com/home-assistant/core/tree/712492b0665240d933206d402d5583fad2bd60ab/homeassistant/components/xiaomi_miio)

[Xiaomi miio Hame assistant](https://www.home-assistant.io/integrations/xiaomi_miio/#xiaomi-mi-robot-vacuum)

**Comment by MTrab** [comment]
Okay - that integration are not using the new target attributes in services.yaml, so that's the difference.
The Miio integration have a field called entity_id, that's not the new standard way ;)

So you need to figure out how to set a target in the javascript

**Comment by Barma-lej** [comment]
But, call other services e.g. partymode, start, lock, without additionals options (zone: 1), are working well

**Comment by MTrab** [comment]
Hmmm - will need to do some testing then.
Is it your own landroid card you have this issue in?

**Comment by Barma-lej** [comment]
Yes, the card is forked from vacuum card. I have seen a several cards and all of these cards call the services as described [here](https://developers.home-assistant.io/docs/frontend/data/#hasscallservicedomain-service-data)

**Comment by MTrab** [comment]
Doesn't seem like zone calls do anything in v0.1.9 - nothing happens, not even an error.

**Comment by MTrab** [comment]
Okay - that makes debugging pretty difficult :D

**Comment by Barma-lej** [comment]
Returned, executed with an error. Now you can debug 🙂

**Comment by Barma-lej** [comment]
Fixed in [landroid-card version 0.2.2](https://github.com/Barma-lej/landroid-card/releases/tag/0.2.2)

https://github.com/MTrab/landroid_cloud/blob/a65160f7b385e7c3386a06582ada8d7b49a4b6fd/custom_components/landroid_cloud/services.py#L116
If called service without additional parmeters, e.g. `lock`, `partymode` your python get service_data as an **array**, otherwise as a **string**

![Screenshot 2022-08-01 001151](https://user-images.githubusercontent.com/34010886/182047331-f12a46b4-a48f-4db3-b549-0fea90f5f6b9.jpg)
